### ISSUE TITLE
Coupon Quantity

### ISSUE THREAD

**Issue by edwinlunando** [byteweaver/django-coupons#11]
If I want to limit the quantity of the coupon that can be redeem, how I can achieve that?

**Comment by Akuryou** [comment]
Hi,

What do you mean by that?

Like how often a user can use the same coupon?
Or the actual value a coupon is worth?

**Comment by edwinlunando** [comment]
I mean that what if I do not want to bind a coupon to a User, instead, I want to publish the coupon code and let people use it publicly, but I will limit the quantity of the coupon usage. Let set 10 times. Aftter 10 times, the coupon will expired.

**Comment by Akuryou** [comment]
Ah.
I'm sorry that's not possible currently.
When I first started this app the requirements were coupons may be used by exactly one user.

But it would be more flexible if 1 coupon could be used by more than one users or a given list of users for a specific number of times.

Actually that's a feature I could use in the near future :)

**Comment by Akuryou** [comment]
Oh I even created an issue long ago about this :D

#8

**Comment by ArijitWeird** [comment]
Is the change has been merged yet??

**Comment by noxan** [comment]
This use case is now supported through Coupon.user_limit.

For a publicly shared coupon usable by ten distinct users:

 ```python
   coupon = Coupon.objects.create_coupon(
       "monetary",
       100,
       user_limit=10,
   )
 ```

Each redemption is tracked through CouponUser, and the coupon becomes fully redeemed when the limit is reached. Unlimited overall use, while allowing each identified user only once, is represented by user_limit=0.

 This behavior is covered by the modernized test suite:

 https://github.com/byteweaver/django-coupons/blob/main/coupons/tests/test_use_cases.py

 Closing as implemented. Please open a new issue if the current behavior differs from the original requirement.
